os/fs/smartfs/smartfs_utils.c

```diff
@@ -1223,10 +1242,11 @@ int smartfs_find_availableentry(struct smartfs_mountpt_s *fs, uint16_t *parentdi
 		psector = nextsector;
 	}
 
-	*parentdirsector = direntry->dsector;
+	/* Retain last active sector in current parent directory to comapre while writing new entry */
+	direntry->last_dsector = direntry->dsector;
 	if (ret != OK) {
 		/* No available entry was found, so we create one */
-		ret = smartfs_createentry(fs, *parentdirsector, direntry);
+		ret = smartfs_createdirentry(fs, direntry);
```

Comment:
actually line 1247, ret is always -ENOENT, we can remove 'if'

Reply:
Actually it is not always -ENOENT beacuse of line
`ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);`
Which may change the value of ret

But yes, if we had found an entry, function would already have returned,
So here we can avoid the check.